tests/Fixer/ClassNotation/SelfStaticAccessorFixerTest.php

```diff
@@ -414,6 +414,55 @@ public static function baz(mixed $other): void
 }
 ',
         ];
+
+        yield 'enum with nested anonymous class' => [
+            '<?php
+                enum Suit: int implements SomeIntInterface, Z
+                {
+                    case Hearts = 1;
+                    case Clubs = 3;
+                    public const HEARTS = self::Hearts;
+
+                    public function Foo(): string
+                    {
+                        return self::Hearts->Bar()->getBar() . self::class . self::Clubs->value;
+                    }
+
+                    public function Bar(): object
+                    {
+                        return new class {
+                            public function getBar()
+                            {
+                                return self::class;
+                            }
+                        };
+                    }
+                }
+            ',
+            '<?php
+                enum Suit: int implements SomeIntInterface, Z
+                {
+                    case Hearts = 1;
+                    case Clubs = 3;
+                    public const HEARTS = self::Hearts;
```

Comment:
Shouldn't it be:
```suggestion
                    public const HEARTS = static::Hearts;
```
?

Reply:
I think `self` is always preferred in an `enum` over `static`, so not sure if I understand correctly?

Reply:
Yes, but this is input and other places have `static` which is fixed later. I am wondering if you overlooked here or is it different on purpose 🙂.

Reply:
I think you cannot have `static const` (in an enum).
https://3v4l.org/GgVkC#v8.2.6  but let me know if you want another test case! :)

Reply:
Yeah, it ends up with an error. So basically this line is not needed here, since it's not fixed and this const is not used anywhere?

Reply:
maybe it is good to test the rule doesn't break on this code?